servers/vm/shm.c

```diff
@@ -13,26 +13,33 @@ int shm_check_available() {
     return i;
 }
 
-int shm_create(size_t size, paddr_t paddr, int slot) {
-    shared_mems[slot].inuse = true;
-    shared_mems[slot].shm_id = slot;
-    shared_mems[slot].len = size;
-    shared_mems[slot].paddr = paddr;
-    return slot;
+error_t shm_create(struct task* task, size_t size, int* slot) {
+    *slot = shm_check_available();
+    if (slot < 0) {
+        return ERR_UNAVAILABLE;
+    }
+    paddr_t paddr;
```

Comment:
`paddr` should be initialized with zero. `alloc_phy_pages` allows specifying the physical memory page to be mapped if paddr is not zero.
```
paddr_t paddr = 0;
```